From 72c25bbb761de2b2acd9f8b652d63e2a2f1caeed Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@ideasonboard.com>
Date: Mon, 11 Sep 2023 12:17:25 +0300
Subject: [PATCH 0991/1016] media: i2c: imx477: Fix locking in
 imx477_init_controls()

The driver does not lock the imx477 mutex when calling
imx477_set_framing_limits(), leading to:

WARNING: CPU: 3 PID: 426 at drivers/media/v4l2-core/v4l2-ctrls-api.c:934 __v4l2_ctrl_modify_range+0x1a0/0x210 [
videodev]

Fix this by taking the lock.

Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ideasonboard.com>
---
 drivers/media/i2c/imx477.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/i2c/imx477.c b/drivers/media/i2c/imx477.c
index 9bf3048670ff..73c56ad5a8c4 100644
--- a/drivers/media/i2c/imx477.c
+++ b/drivers/media/i2c/imx477.c
@@ -2069,9 +2069,13 @@ static int imx477_init_controls(struct imx477 *imx477)
 
 	imx477->sd.ctrl_handler = ctrl_hdlr;
 
+	mutex_lock(&imx477->mutex);
+
 	/* Setup exposure and frame/line length limits. */
 	imx477_set_framing_limits(imx477);
 
+	mutex_unlock(&imx477->mutex);
+
 	return 0;
 
 error:
-- 
2.42.0

